src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        }
+        size++;
```

Comment:
```suggestion
        Node<T> currentNode = new Node<>(tail , value, null);
        if (head == null) {
            head = currentNode;
        } else {
            tail.next = currentNode;
        }
        tail = currentNode;
        size++;
```
 you can simplify this